### PR TITLE
Setup serverless to deploy elastic and basic lambda

### PR DESCRIPTION
**Pull request by barlock**
**Related Issue**  
Supports #1 

**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
Basic setup of a lambda deployment that can access an already deployed elasticsearch cluster. 

**Description of Changes**  
I created a new set of packages in `services`. Serverless likes to want every service to be it's own actual node package, which actually helps a lot in deployment. BUT, I didn't feel like each service should live in a package as it's not something we _could_ eventually publish, it's ruhl specific to us. SO, I made a new `services` folder that lerna manages 🤷‍♂️. Not sure how it will fair, but I'm enjoying it so far.

I'll leave comments on the weird bits.

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: services/collector/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      Resource: '*'
+      Action:
+        - 'es:ESHttp*'
+
+functions:
+  collect:
+    handler: lambda.handler
+
+    environment:
+      ELASTIC_ENDPOINT: ${cf:collector-elasticsearch-${self:provider.stage}.ElasticsearchDomainEndpoint}
```
Comment (barlock) [inline services/collector/serverless.yml]: This whole "collector" service is largely not useful right now. This is the only interesting line in the whole thing. Can a lambda know where to post things when I eventually write it?

image

🎉

File: services/elasticsearch/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e
+
+DIRNAME="$(cd -P "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+serverless deploy
```
Comment (barlock) [inline services/elasticsearch/deploy.sh]: Mostly copied from the last one, this is the only difference.

File: services/elasticsearch/serverless.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  runtime: nodejs8.10
+  stage: ${opt:stage, 'dev'}
+  region: us-east-1
+
+custom:
+  esDomainName: w3s-data
+  stackName: ${self:service.name}-${self:provider.stage}
+
+resources:
+  Resources:
```
Comment (barlock) [inline services/elasticsearch/serverless.yml]: This is all copied out of the last cloudformation deely which was deleted. 

Interesting things to note is the serverless-cf-vars things where all `${AWS:*}` becomes `#{AWS:*}`

File: packages/github-collector/package.json [hunk 4]
```diff
@@ -25,6 +25,7 @@
     "dotenv": "^7.0.0",
     "elasticsearch": "^15.4.1",
     "lodash": "^4.17.11",
+    "node-fetch": "^2.3.0",
```
Comment (barlock) [inline packages/github-collector/package.json]: node dependencies are weird. 

`serverless` added a 1.x version of node-fetch which somehow overwrote octokit's ask for a 2.x one and broke all the tests 🤷‍♂️ 



Lockin' it in.

File: services/collector/serverless.yml (hunk 1 above)
Comment (breakpointer) [reply services/collector/serverless.yml]: W00T

File: services/elasticsearch/serverless.yml (hunk 3 above)
Comment (breakpointer) [reply services/elasticsearch/serverless.yml]: oh boy... I love situations where the difference between 😄  and 😡 are the pixel diff between `$` and `#`

**Pull request merged**